libs/platform/src/lib/table/table.component.ts

```diff
@@ -1853,22 +1859,24 @@ export class TableComponent<T = any> extends Table implements AfterViewInit, OnD
 
     /** @hidden */
     private _initializeDS(dataSource: FdpTableDataSource<T>): void {
-        if (isDataSource(this._tableDataSource)) {
-            this._closeDataSource(this._tableDataSource);
-        }
-
+        this._closeDataSource();
         this._resetAllSelectedRows();
 
         this._tableDataSource = this._openDataStream(dataSource);
     }
 
     /** @hidden */
-    private _closeDataSource(dataSource: TableDataSource<T>): void {
-        dataSource.close();
+    private _closeDataSource(): void {
+        if (!this._tableDataSource) {
+            return;
+        }
 
-        this._subscriptions.remove(this._dsSubscription);
+        this._tableDataSource.close();
 
         if (this._dsSubscription) {
+            this._subscriptions.remove(this._dsSubscription);
+
+            this._dsSubscription.unsubscribe();
```

Comment:
do we need to `_subscriptions.remove` after `this._dsSubscription.unsubscribe(); `?
Doesn't remove happen automatically in this case?

Reply:
Yea, we can, thank you.